simulator/modules/decode/decode.h

```diff
@@ -56,6 +60,29 @@ class Decode : public Log
         void clock( Cycle cycle);
         void set_RF( RF<FuncInstr>* value) { rf = value;}
         void set_wb_bandwidth( uint32 wb_bandwidth) { bypassing_unit->set_bandwidth( wb_bandwidth);}
+        auto get_target_mispredictions_num() const { return num_target_mispredictions; }
+        auto get_direction_mispredictions_num() const { return num_direction_mispredictions; }
+        auto get_mispredictions_num() const { return num_target_mispredictions + num_direction_mispredictions; }
+        auto get_jumps_num() const { return num_jumps; }
+
+        bool is_misprediction( const Instr& instr, const BPInterface& bp_data)
+        {
+            if ( ( instr.is_direct_jump() || instr.is_indirect_jump() || instr.is_likely_branch()) && !bp_data.is_taken)
+            {
+                num_direction_mispredictions++;
```

Comment:
Mixing mutating (num_direction_mispredictions) and accessing (is_misprediction) behavior in a single function is a bad practice. Please leave `is_misprediction` as an accessing method with const qualifier.